crates/tuono_lib/src/request.rs

```diff
@@ -68,6 +80,42 @@ impl Request {
             "No body found",
         )))
     }
+
+    pub fn form_data<T>(&self) -> Result<T, FormError>
+    where
+        T: DeserializeOwned,
+    {
+        let content_type = self
+            .headers
+            .get("content-type")
+            .and_then(|v| v.to_str().ok())
+            .unwrap_or("");
+
+        if !content_type.contains("application/x-www-form-urlencoded") {
+            return Err(FormError::InvalidContentType);
+        }
+
+        let body_str = match self.body_as_string() {
+            Ok(s) => s,
+            Err(e) => return Err(FormError::ParseError(e)),
+        };
+
+        let form_data = match serde_json::to_value(
+            form_urlencoded::parse(body_str.as_bytes())
+                .into_owned()
+                .collect::<HashMap<String, String>>(),
+        ) {
```

Comment:
I think this is inefficient. We translate first the the bytes into a string (cloning the boty) and then again into bytes.

Couldn't we just read the bytes?